### PR TITLE
Create Confluent Kafka Schema Registry Chart

### PR DESCRIPTION
**Pull request by benjigoldberg**
Hello Maintainers!

This PR adds the [Confluent Schema Registry](https://github.com/confluentinc/schema-registry) as an incubator chart. Confluent Schema Registry is used alongside a Kafka installation to gracefully manage Avro Schema versions. The server itself may be scaled up and down gracefully via the traditional `ReplicaCount` Deployment value. This chart requires a running Kafka and Zookeeper installation, either in-cluster  or out-of-cluster to function. By default with no arguments this chart will install a single Kafka and Zookeeper pod.

Finally, I think its worth discussing whether the functionality in this chart is best left independent on its own, or as an optional installation that is part of the Kafka Chart. I waffled on this decision for a bit, but ultimately decided that allowing users to install it standalone makes sense, since the majority of Kafka installations in place today are not currently running in Kubernetes and it would be nice to offer this as a stand-alone feature. Note that I have added this chart as an optional requirement to the Kafka chart as well, but it is disabled by default.

### CODE REVIEW

**Comment by benjigoldberg** [comment]
@viglesiasce we spoke a few weeks ago in the sig-charts weekly video conference about general Kafka improvements -- just wanted to give you a heads up that this will be my first in a series of PRs intended to extend the functionality of Kafka and its general ecosystem within helm/k8s

**Comment by viglesiasce** [comment]
Thanks @benjigoldberg! Can you do me a favor and split this one into 2 parts:
1. The new schema-registry chart
2. The update to use it in the kafka chart

**Comment by benjigoldberg** [comment]
@viglesiasce done -- Kafka changes are now in PR #2887

File: incubator/schema-registry/values.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Default values for Confluent Schema-Registry
+# This is a YAML-formatted file.
+# Declare name/value pairs to be passed into your templates.
+# name: value
+
+## schema-registry repository
+Image: "confluentinc/cp-schema-registry"
```
Comment (viglesiasce) [inline incubator/schema-registry/values.yaml]: We prefer these to be in camelCase

**Review by viglesiasce — CHANGES_REQUESTED** [review]
Please make values camelCase. Otherwise looks good.

**Comment by benjigoldberg** [comment]
@viglesiasce updated with the exception of the kafka chart variable overrides -- its using upper `CamelCase` for `Replicas` and `zookeeper.Servers` -- I'll change those vars to lower `camelCase` in a future kafka-centric PR if thats ok w/ you

**Comment by viglesiasce** [comment]
@benjigoldberg sounds great!

**Review by viglesiasce — APPROVED** [review]

**Pull request merged**